### DISCUSSION TITLE
can transmit errors

### DISCUSSION THREAD

**Discussion by mo22** [Neuroquila-n8fall/JunkersControl#41] [Q&A] [answered]
Hello,

first of all, thank you for this amazing project.

I'm able to receive the can bus data, but transmitting (for getting rid of the ta300) seems to cause an error.

I've connected a different MCP2515 board ("niren" style) to an arduino and tried again - again I can receive, but not transmit.

Transmitting fails with the TXERR & TXREQ bits set, which, as i understand the documentation, means that the transmission was not acknowledged.

I've tried this with and without the TA300 on the bus, with and without terminating resistors, with and without connecting GND together.

I've also tried an external power supply to make sure the mcp2515 gets really 5V (and not some USB 4.8xV).

All transmissions fail (0x00f9 / 0x0250-0x0258).

Now I'm not sure what I'm missing here, what setup / cabling are you using? (especially with the esp32 - level shifter?)

Thank you very much

Mo

**Comment by mo22** [comment, marked as answer]
So, I found something.

The CAN transceiver (TJA1050 on most boards, but also MCP2551) does not support transmitting in 10 kbps. The issue is the "TXD dominant time for time-out" of typically 450 us, and each bit time having 100 us and a maximum repetition of 5 (can bit stuffing) ends up at 500 us which triggers an error.

Maybe the author of this project got lucky with the MCP board, as the datasheet says 450 us typical, 250 us minimum.

**Reply by Neuroquila-n8fall** [reply]
Well...good find! That explains a lot. I have like 12 of those modules here and only some of them appear to work properly. I ended up using a completely different module: Debo / Joy-It

**Reply by Neuroquila-n8fall** [reply]
oh - on another note Joe found a way of attaching directly to the heating controller via I2C, rendering the bus module and can module obsolete - but it has it's own challenges, of course.

**Comment by mo22** [comment]
Yes, I also got a Debo module has the mcp2562 transceiver, which, according to its datasheet, supports down to 16 kbps, so you still have to get kind of lucky.

I also found that it is quite simple to bitbang transmit at that data rate, using a timer and two transistors to pull H/L high and low, as well as four resistors to tie H/L to 5V/gnd to have them at 2.5V idle.

But the Debo board seems to be the easiest way.
